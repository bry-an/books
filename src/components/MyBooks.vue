<template>
  <div>
    <h2>My Selected Books</h2>
    <v-data-table
      no-data-text="You don't have any books selected yet"
      :headers="headers"
      :items="books"
    >
      <template v-slot:items="props">
        <td class="text-xs-center">
          <v-icon @click="$emit('remove-item', props.item)" small>remove_circle</v-icon>
        </td>
        <td class="text-xs-left">{{props.item.title}}</td>
        <td class="text-xs-left">{{ props.item.author_name[0] }}</td>
        <td class="text-xs-left">{{ props.item.first_publish_year}}</td>
        <td class="text-xs-left pa-0">
          <v-btn class="mr-3" primary small @click="routeToViewBook(props.item)">More Info</v-btn>
        </td>
      </template>
    </v-data-table>
  </div>
</template>

<script>
export default {
  name: "MyBooks",
  data() {
    return {
      headers: [
        {
          text: "Remove",
          sortable: false
        },
        {
          text: "Title",
          value: "title"
        },
        {
          text: "Author",
          value: "author_name[0]"
        },
        {
          text: "First published",
          value: "first_publish_year"
        },
        {
          text: "More info"
        }
      ]
    };
  },
  props: {
    books: {
      type: Array
    }
  },
  methods: {
    routeToViewBook(book) {
      this.$router.push({ name: "BookView", params: book });
    }
  }
};
</script>

<style lang="scss" scoped>
</style>