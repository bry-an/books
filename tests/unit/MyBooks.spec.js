import { mount } from "@vue/test-utils"
import Vuetify from "vuetify"
import MyBooks from "../../src/components/MyBooks.vue"
import Vue from "vue"
Vue.use(Vuetify)

describe("My Books", () => {
    let build
    build = () => {
        const books = [{
            title: "",
            author_name: [""],
            first_publish_year: 0
        }]

        const wrapper = mount(MyBooks, {
            propsData: {
                books
            }
        })
        return {
            wrapper
        }
    }
    it("renders the component", () => {
        const { wrapper } = build()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it("displays books", () => {
        const { wrapper } = build()
        const td = wrapper.find("td")
        expect(td.exists()).toBe(true)
    })
})