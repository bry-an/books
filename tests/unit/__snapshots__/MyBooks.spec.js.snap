// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Books renders the component 1`] = `
<div>
  <h2>My Selected Books</h2>
  <div>
    <div class="v-table__overflow">
      <table class="v-datatable v-table theme--light">
        <thead>
          <tr>
            <th role="columnheader" scope="col" aria-label="Remove: Not sorted." aria-sort="none" class="column text-xs-left">Remove</th>
            <th role="columnheader" scope="col" aria-label="Title: Sorted ascending. Activate to sort descending." aria-sort="ascending" tabindex="0" class="column sortable active asc text-xs-left">Title<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="Author: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Author<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="First published: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">First published<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
            <th role="columnheader" scope="col" aria-label="More info: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">More info<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
          </tr>
          <tr class="v-datatable__progress">
            <th colspan="5" class="column"></th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td class="text-xs-center"><i aria-hidden="true" class="v-icon v-icon--link material-icons theme--light" style="font-size: 16px;">remove_circle</i></td>
            <td class="text-xs-left"></td>
            <td class="text-xs-left"></td>
            <td class="text-xs-left">0</td>
            <td class="text-xs-left pa-0"><button type="button" class="mr-3 v-btn v-btn--small theme--light" primary="">
                <div class="v-btn__content">More Info</div>
              </button></td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-datatable v-table theme--light">
      <div class="v-datatable__actions">
        <div class="v-datatable__actions__select">Rows per page:<div role="combobox" class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light">
            <div class="v-input__control">
              <div class="v-input__slot">
                <div class="v-select__slot">
                  <div class="v-select__selections">
                    <div class="v-select__selection v-select__selection--comma">5</div><input aria-label="Rows per page:" readonly="readonly" type="text" autocomplete="on" aria-readonly="false">
                  </div>
                  <div class="v-input__append-inner">
                    <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_down</i></div>
                  </div>
                </div>
                <div class="v-menu">
                  <div class="v-menu__content theme--light v-menu__content--auto " style="max-height: 200px; min-width: 75px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;">
                    <div class="v-select-list v-card theme--light">
                      <div role="list" class="v-list theme--light">
                        <div role="listitem" class="primary--text"><a class="v-list__tile v-list__tile--link v-list__tile--active theme--light primary--text">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">5</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">10</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">25</div>
                            </div>
                          </a></div>
                        <div role="listitem"><a class="v-list__tile v-list__tile--link theme--light">
                            <div class="v-list__tile__content">
                              <div class="v-list__tile__title">All</div>
                            </div>
                          </a></div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class="v-datatable__actions__range-controls">
          <div class="v-datatable__actions__pagination">1-1 of 1</div><button disabled="disabled" type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light" aria-label="Previous page">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_left</i></div>
          </button><button disabled="disabled" type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light" aria-label="Next page">
            <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_right</i></div>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
